polynote-frontend/polynote/ui.js

```diff
@@ -1964,17 +1972,36 @@ export class MainUI extends EventTarget {
     }
 
     createNotebook(evt) {
-        const notebookPath = prompt("Enter the name of the new notebook (no need for an extension)");
-        if (notebookPath) {
-            const handler = this.socket.addMessageListener(messages.CreateNotebook, (actualPath) => {
-                this.socket.removeMessageListener(handler);
-                this.browseUI.addItem(actualPath);
-                this.loadNotebook(actualPath);
-            });
+        const notebookPath = prompt("Enter the name of the new notebook (no need for an extension), or the full URL of another Polynote instance.");
 
+        const handler = this.socket.addMessageListener(messages.CreateNotebook, (actualPath) => {
+            this.socket.removeMessageListener(handler);
+            this.browseUI.addItem(actualPath);
+            this.loadNotebook(actualPath);
+        });
+
+        if (notebookPath.startsWith("http")) {
```

Comment:
Is this for importing a foreign notebook? Won't you run into cross-domain issues doing this from the client?

Reply:
Oops, good call! Had to add the magic CORS headers and it works: https://github.com/polynote/polynote/pull/225/commits/f2ad4b3cf688dd3a2046b193577c3402501c132d

I confess I don't 100% know all the security implications behind just openin' everything up like this though so lmk if you know of a better way!

Reply:
Could have the server do the downloading?